packages/harpguru-core/src/hooks/use-viewable-matrices/use-viewable-matrices.test.ts

```diff
@@ -0,0 +1,197 @@
+import { useGlobal } from 'reactn'
+import { DegreeIds, InteractionIds, PitchIds } from 'harpparts'
+
+import { inactiveCellsHarpStrata } from '../../test-resources'
+
+import { useViewableMatrices } from './use-viewable-matrices'
+
+jest.mock('reactn')
+const mockUseGlobal = useGlobal as jest.Mock
+
+test('Entire degreeMatrix is viewable when columnBounds is FIT', () => {
+  mockUseGlobal.mockImplementation((stateItem: string) => {
+    if (stateItem === 'activeHarpStrata') return [inactiveCellsHarpStrata]
+    if (stateItem === 'columnBounds') return ['FIT']
+    return undefined
+  })
+
+  expect(useViewableMatrices().viewableDegreeMatrix).toStrictEqual(
+    inactiveCellsHarpStrata.degreeMatrix
+  )
+})
+
+test('First column of degreeMatrix is viewable when columnBounds is [0, 0]', () => {
+  const { Root, Second, Third } = DegreeIds
+  // TODO: This is actually subverting the type checking.
```

Comment:
TODO here

Reply:
Raised as techdebt in https://github.com/js-jslog/harpguru/issues/113#issuecomment-922351622

The work done to make the app sizing dynamic (at #107) was used as a platform for additional speculative pull requests to ensure that this was the correct direction to start down.

Rather than addressing the finer details in this or subsequent pull request and potentially struggling to merge up to those subsequent branches I will prefer to record the issues in each merge request and resolve them afterwards.

This has a limit of course. Major issues will not be merged to master. I will need to find a way around that if it happens.